### PR TITLE
Refactor, new features, encapsulation, docstrings(Sphinx)

### PR DESCRIPTION
**Pull request by meyafex**
I've made a quick guide for users using sphinx. refactored code, tested it.

### CODE REVIEW

**Comment by meyafex** [comment]
to check how html doc looks like:
pip install Sphinx
cd docs
make html
open _build/html/index.html

File: bpx/bpx.py [hunk 1]
```diff
@@ -6,125 +6,311 @@
...
+        :type proxies: :class:`dict`, optional
+        .. _requests documentation: https://requests.readthedocs.io/en/latest/user/advanced/#proxies
+
+    """
+    _URL = 'https://api.backpack.exchange/'
+    debug = False
+    debugTs = 0
 
-    def init(self, api_key, api_secret):
+    def __init__(self, api_key: str, api_secret: str, window: int = 5000, proxies: dict = None):
```
Comment (meyafex) [inline bpx/bpx.py:26]: more OOP and understandable class initialization

Comment (sndmndss) [inline bpx/bpx.py:26]: more OOP and understandable class initialization

File: bpx/bpx.py [hunk 2]
```diff
@@ -6,125 +6,311 @@
...
+        :return: Account balances and the state of the balances (locked or available).
+        :rtype: :class:`dict`
+        """
+        return self._handle_bpx_request(url=f'{self._URL}api/v1/capital',
+                                        headers=self._sign('balanceQuery'))
 
-    def deposits(self):
-        return requests.get(url=f'{self.url}wapi/v1/capital/deposits', proxies=self.proxies,
-                            headers=self.sign('depositQueryAll', {})).json()
+    def deposits(self, limit: int = 100, offset: int = 0):
```
Comment (meyafex) [inline bpx/bpx.py:55]: new params, tested

File: bpx/bpx.py [hunk 3]
```diff
@@ -6,125 +6,311 @@
...
+        """
+        params = {'blockchain': chain}
+        return self._handle_bpx_request(url=f'{self._URL}wapi/v1/capital/deposit/address',
+                                        headers=self._sign('depositAddressQuery', params),
+                                        params=params)
 
     # set withdrawal address:
     # https://backpack.exchange/settings/withdrawal-addresses?twoFactorWithdrawalAddress=true
-    def withdrawal(self, address: str, symbol: str, blockchain: str, quantity: str):
+    def get_withdrawals(self, limit: int = 100, offset: int = 0):
```
Comment (meyafex) [inline bpx/bpx.py:87]: better naming

File: bpx/bpx.py [hunk 4]
```diff
@@ -6,125 +6,311 @@
...
+        :rtype: :class:`dict`
+        """
         params = {'symbol': symbol}
-        return requests.delete(url=f'{self.url}api/v1/orders', proxies=self.proxies, data=json.dumps(params),
-                               headers=self.sign('orderCancelAll', params)).json()
+        return self._handle_bpx_request(url=f'{self._URL}api/v1/orders', params=params,
+                                        headers=self._sign('orderCancelAll', params), r_type='DELETE')
 
-    def sign(self, instruction: str, params=None):
+    def _sign(self, instruction: str, params=None):
```
Comment (meyafex) [inline bpx/bpx.py:312]: user not supposed to see this method in his IDE, as all methods below

**Pull request closed without merging**